src/components/Tool/Appeal/AppealDetails/AppealHeaderInfo/AppealHeaderInfo.test.tsx

```diff
@@ -54,5 +58,26 @@ describe('AppealHeaderInfo', () => {
     });
   });
 
-  // TODO - Build tests for modals opening and saving data
+  it('should allow user to open the edit appeal info modal', async () => {
+    const { getByText, getByTestId, getByRole } = render(
+      <Components appealInfo={appealInfo} loading={false} />,
+    );
+
+    await waitFor(() => {
+      expect(getByText('Test Appeal')).toBeInTheDocument();
+    });
+
+    userEvent.click(getByTestId('edit-appeal-name'));
+    await waitFor(() => {
+      expect(getByRole('heading', { name: 'Edit Appeal' })).toBeInTheDocument();
+    });
+
+    const button = getByRole('button', { name: 'Close' });
+    within(button).getByTestId('CloseIcon');
```

Comment:
Do you want to assert that the `CloseIcon` is in the document?

Reply:
No, I just want to press the correct close button as I know the filters are open, so there will be 2 close buttons in the HTML

Reply:
Ah, I see. I've fixed it by editing the test